### PR TITLE
Inference speed ups

### PR DESCRIPTION
**Pull request by WilliamIX**
This PR refactors some of the inference code and optimizes the norm computation to use custom cython code instead of numpy code. This closes issue #54 .

refactorization consists of pre-instantiating most likelihood objects (and data objects) as opposed to instantiating them every iteration of the cost function.

The number of walkers used in the examples has also been lowered to 5-10.

Ultimately, all the deterministic examples now run in 30 seconds or less. Stochastic examples may take a couple of minutes depending on the number of simulations to conduct per parameter sample.

The changes to simulator.pyx were put there for debugging purposes, but ultimately the bug was user-error. However, those functions are still helpful for debugging.

### CODE REVIEW

File: bioscrape/sbmlutil.py [hunk 1]
```diff
@@ -65,7 +65,11 @@ def import_sbml(sbml_file, bioscrape_model = None, input_printout = False, **kwa
     for reaction in model.getListOfReactions():
         # get the propensity
         kl = reaction.getKineticLaw()
+
+
         # capture any local parameters
+        # also must save renamed local parameters to rename annotations later
+        renamed_params = {}
```
Comment (ayush9pandey) [inline bioscrape/sbmlutil.py:72]: Seems like the changes from the previous PR are not merged/in sync yet? Why are the changes in sbmlutil on removing underscores from parameters showing up here?

**Review by ayush9pandey — CHANGES_REQUESTED** [review]
Other than the old SBML changes that appear in this PR, rest of it seems fine to me.

**Comment by ayush9pandey** [comment]
Also wanna make sure if you checked that all the examples in "inference examples" work? Let me know if you want me to check that.

File: bioscrape/sbmlutil.py (hunk 1 above)
Comment (ayush9pandey) [reply bioscrape/sbmlutil.py:72]: Seems like you created this PR to master, maybe that's the reason?

Comment (WilliamIX) [reply bioscrape/sbmlutil.py]: That was just a mistake - changed the merge into dev instead. Everything look okay?

**Comment by WilliamIX** [comment]
> Also wanna make sure if you checked that all the examples in "inference examples" work? Let me know if you want me to check that.

I did check all of them - but it wouldn't be bad if you looked too because you know what they "should" look like better than I do.

**Review by ayush9pandey — APPROVED** [review]

**Comment by ayush9pandey** [comment]
I changed the base to dev, so these changes will be merged to dev first, since I also want to make a few changes to inference module. Once that's done, we can push to master again.

**Pull request merged**